bioblend/_tests/TestGalaxyJobs.py

```diff
@@ -146,3 +146,86 @@ def test_rerun_and_remap(self):
         self.assertEqual(last_dataset['hid'], 3)
         self.assertEqual(last_dataset['id'], history_contents[2]['id'])
         self._wait_and_verify_dataset(last_dataset['id'], b'line 1\tline 1\n')
+
+    @test_util.skip_unless_galaxy('release_19.05')
+    @test_util.skip_unless_tool("random_lines1")
+    def test_get_common_problems(self):
+        job_id, _ = self._run_tool()
+        response: dict = self.gi.jobs.get_common_problems(job_id)
+        self.assertEqual(response, {'has_duplicate_inputs': False, 'has_empty_inputs': True})
+
+    @test_util.skip_unless_tool("random_lines1")
+    def test_get_inputs(self):
+        job_id, _ = self._run_tool()
+        response: dict = self.gi.jobs.get_inputs(job_id)
+        self.assertEqual(response, [{'name': 'input', 'dataset': {'src': 'hda', 'id': self.dataset_id}}])
+
+    @test_util.skip_unless_tool("random_lines1")
+    def test_get_outputs(self):
+        job_id, output_id = self._run_tool()
+        response: dict = self.gi.jobs.get_outputs(job_id)
+        self.assertEqual(response, [{'name': 'out_file1', 'dataset': {'src': 'hda', 'id': output_id}}])
+
+    @test_util.skip_unless_galaxy('release_18.09')
+    @test_util.skip_unless_tool("random_lines1")
+    def test_resume_job(self):
+        job_id, output_id = self._run_tool()
+        response: dict = self.gi.jobs.resume_job(job_id)
+        self.assertEqual(response, [{'name': 'out_file1', 'dataset': {'src': 'hda', 'id': output_id}}])
+
+    @test_util.skip_unless_galaxy('release_20.05')
+    @test_util.skip_unless_tool("random_lines1")
+    def test_get_destination_params(self):
+        job_id, _ = self._run_tool()
+        response: dict = self.gi.jobs.get_destination_params(job_id)
+        self.assertIn('Runner', response)
+        self.assertIn('Runner Job ID', response)
+        self.assertIn('Handler', response)
+
+    @test_util.skip_unless_tool("random_lines1")
+    def test_search_jobs(self):
+        tool_id = 'random_lines1'
+        job_id, _ = self._run_tool(tool_id=tool_id)
+        response: dict = self.gi.jobs.search_jobs({'tool_id': tool_id, 'inputs': self.gi.jobs.get_inputs(job_id)[0]})
+        self.assertEqual(response, [])
```

Comment:
I think this test is not correct yet, because `search_ jobs` should not return `[]` since it should at least find the job corresponding to `job_id`.

Reply:
Try this:

```
        inputs = {
            'num_lines': '1',
            'input': {
                'src': 'hda',
                'id': self.dataset_id
            },
            'seed_source|seed_source_selector': 'set_seed',
            'seed_source|seed': 'asdf'
        }
        response: dict = self.gi.jobs.search_jobs({'tool_id': tool_id, 'inputs': inputs})
```

There are two problems: 1) the get_inputs method is only designed to get the dataset inputs, whereas the search requires that all params are identical, and 2) that using `input_format='21.01'` isn't supported by the search API, so we need to enter the params using the `legacy` format with the `|` rather than using nested dicts.

Reply:
Thank you for the fix, and the helpful additional information